controlplane/internal/controller/clusterdeployment_controller.go

```diff
@@ -273,3 +281,41 @@ func (r *ClusterDeploymentReconciler) createImageRegistry(ctx context.Context, r
 	}
 	return nil
 }
+
+// setACICapabilities will set the install config override annotation in the AgentClusterInstall if there
+// are additional enabled capabilities that need to be defined.
+// For SNO (single node openshift) and non-baremetal platform MNO (multi-node openshift), this is set if
+// the OpenshiftAssistedControlPlane has additional capabilities set.
+// For MNO (multi-node openshift), this is set to the default list for baremetal platform,
+// but the OpenshiftAssistedControlPlane can specify additional capabilities to be appended to this list.
+func setACICapabilities(oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane, aci *hiveext.AgentClusterInstall) error {
+	isSNO := oacp.Spec.Replicas == 1
+	isBaremetalPlatform := aci.Spec.PlatformType == hiveext.BareMetalPlatformType
+
+	additionalCapabilitiesList := []string{}
+	if !isSNO && isBaremetalPlatform {
+		additionalCapabilitiesList = append([]string{}, defaultMNOAdditionalCapabilities...)
+	}
+
+	for _, capability := range oacp.Spec.Config.AdditionalEnabledCapabilities {
+		// Ignore MAPI for baremetal MNO clusters and ignore duplicates
+		if (strings.EqualFold(capability, "MachineAPI") && !isSNO && isBaremetalPlatform) || funk.ContainsString(additionalCapabilitiesList, capability) {
```

Comment:
I think we do not really care if this is SNO or not, right? we only care about the platform? Is MAPI capability added only for baremetal platform?

Reply:
> I think we do not really care if this is SNO or not, right? we only care about the platform?

Oh yes good catch 😆 

> Is MAPI capability added only for baremetal platform?

MAPI should not be added for baremetal platform